agent/03_agent_deployment.sh

```diff
@@ -107,17 +119,70 @@ stringData:
   .dockerconfigjson: '${pull_secret}'
 
 EOF
+
+    # Get the generated mac addresses to set up nmstate
+    mac_list=($(sudo virsh net-dumpxml ${BAREMETAL_NETWORK_NAME} | grep -Po '(?<=host mac=).*' | cut -d' ' -f1))
+    ip_list=$2
+
+    add_host_dns_entries ${ip_list}
+
+    num_ips=${#ip_list[@]}
+
+    echo -n "" > ${FLEETING_MANIFESTS_PATH}/nmstateconfig.yaml
+
+    # Create a yaml for each host in nmstateconfig.yaml
+    for (( i=0; i<$num_ips; i++ ))
+    do
+        cat >> "${FLEETING_MANIFESTS_PATH}/nmstateconfig.yaml" << EOF
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: mynmstateconfig
+  namespace: openshift-machine-api
+  labels:
+    cluster0-nmstate-label-name: cluster0-nmstate-label-value
+spec:
+  config:
+    interfaces:
+      - name: eth0
+        type: ethernet
+        state: up
+        mac-address: ${mac_list[i]}
+        ipv4:
+          enabled: true
+          address:
+            - ip: ${ip_list[i]}
+              prefix-length: 24
```

Comment:
Here you could use `CLUSTER_HOST_PREFIX`

Reply:
Good idea, changed to use that.